### PR TITLE
fix: harden token revocation, playground link handling, and dependency

### PR DESCRIPTION
**Pull request by ThomasRooney**
Fixes [SEC-42](https://linear.app/speakeasy/issue/SEC-42)

---
## Summary by cubic
Hardens token revocation and MCP playground open-link handling to address SEC-42. We now verify HS256 signatures before using JWT jti and only allow http/https links; added tests and bumped `anyio`.

- **Bug Fixes**
  - Revoke: verify HS256 signature (exact alg pinning) before extracting jti, tolerate expired tokens, and revoke only when the session belongs to the authenticated client.
  - JWT: add VerifiedJTI and pin HS256 in Validate; tests cover valid, expired, wrong-key, alg=none/HS384, missing-jti, and malformed tokens.
  - Playground: allowlist http/https before `window.open` for MCP `ui/open-link`.

- **Dependencies**
  - Bumped `anyio` from 4.14.1 to 4.14.2.

<sup>Written for commit 3eda4646bf48a632cb18f47866e7dbe41d448406. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/speakeasy-api/gram/pull/4852?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: client/dashboard/src/pages/playground/PlaygroundMcpApps.tsx [hunk 1]
```diff
@@ -571,6 +574,18 @@ async function requestMcpJsonRpc<T>(
   return payload.result as T;
 }
 
+// Links surfaced by an embedded MCP app are untrusted input; only allow
+// navigation to web URLs so a frame cannot open javascript:, file:, or
+// arbitrary custom-scheme links from the host page.
+function isSafeExternalUrl(raw: string): boolean {
```
[the remark this answers is not in the document]
Comment (ThomasRooney) [reply client/dashboard/src/pages/playground/PlaygroundMcpApps.tsx:580]: no

**Review by tgmendes — APPROVED** [review]
